lib/main.ts

```diff
@@ -156,9 +156,15 @@ export class Transpiler {
 
   // Visible for testing.
   getOutputPath(filePath: string, destinationRoot: string): string {
-    var relative = this.getRelativeFileName(filePath);
-    var dartFile = relative.replace(/.(js|es6|ts)$/, '.dart');
-    return path.join(destinationRoot, dartFile);
+    var dartFile = filePath.replace(/.(js|es6|ts)$/, '.dart');
```

Comment:
I think it's time to give up here and use some proper library call to make the path relative.

I don't know why we're not using `path.resolve(destinationRoot, relative)` here. Care to try?